In [1]:
import sys
sys.path.append("..")
from pathlib import Path
import plotly.graph_objects as px
from utils.transcription_utils import get_transcription_info,get_stats,get_total_stats
import csv
import pandas as pd
import json

In [2]:
save_path=Path("/media/dataset-harddisk/munikumar/utils/audio_utils/data/ekstep/normalized")

In [3]:
dataset_dir=Path("/media/dataset-harddisk/munikumar/hindi_dataset/ekstep/ekstep_raw/ekstep_benchmark/Hindi_benchmarking_dataset_external_21-03-2022_06-46")

In [4]:
def load_transcription(transcription_path):
    transcription_data={}
    with open(transcription_path) as audio_data_file:
        data = json.load(audio_data_file)
    for audio_data in data:
        transcription_data[audio_data["audioFilename"]]=audio_data["text"]
    return transcription_data

In [5]:
transcription_data=load_transcription(dataset_dir/"data.json")
get_transcription_info(transcription_data,dataset_dir,save_path,"test")
get_stats(save_path/"test_non_hindi_transcripts.json")

getting audio info about : normalized - test


100%|██████████| 11064/11064 [12:48<00:00, 14.39it/s]


**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 20:31:33
Loss duration : 3:54:39
Remaining duration : 16:36:54
Drop by percentage: 19.05

Total audio files : 11064
Files that would be drop: 1820
Remaining audio files : 9244
Drop by percentage: 16.45
**************************************************
Non hindi tokes : ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'B', 'E', 'G', 'H', 'I', 'K', 'M', 'N', 'a', 'b', 'c', 'd', 'e', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'r', 's', 't', 'u', 'v', 'w']


### Test Dataset
**************************************************
Loss of audio data without transcription normalization:
**************************************************
Total duration : 20:31:33  
Loss duration : 4:54:32  
Remaining duration : 15:37:1  
Drop by percentage: 23.92  

Total audio files : 11064  
Files that would be drop: 2093  
Remaining audio files : 8971  
Drop by percentage: 18.92  
**************************************************
Non hindi tokes : ['%', '*', '+', '-', '.', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', ':', 'B', 'E', 'G', 'H', 'I', 'K', 'M', 'N', 'a', 'b', 'c', 'd', 'e', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'r', 's', 't', 'u', 'v', 'w', '₹']

**************************************************
Loss of audio data with transcription normalization:
**************************************************
Total duration : 20:31:33  
Loss duration : 3:54:39  
Remaining duration : 16:36:54  
Drop by percentage: 19.05  

Total audio files : 11064  
Files that would be drop: 1820  
Remaining audio files : 9244  
Drop by percentage: 16.45  
**************************************************
Non hindi tokes : ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'B', 'E', 'G', 'H', 'I', 'K', 'M', 'N', 'a', 'b', 'c', 'd', 'e', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'r', 's', 't', 'u', 'v', 'w']